include/ArcadeMachine.h

```diff
@@ -20,333 +20,467 @@
-            this->_game_btns = menu.get_buttons();
-            
-            while ((!key_typed(ESCAPE_KEY) && !overlayActive) || overlayActive)
-            {
-                //write_line("into while");
-                overlayActive = menu.get_overlay_state();
-                process_events();   
-                clear_screen();
-                // Get mouse position
-                this->_mouse = mouse_position();
-                // Draw games menu
-                menu.draw_menu_page();
-                // Check input
-              //  this->_action = this->_selector_games_menu.check_key_input(this->_games_btns);
-                refresh_screen(60);
-            }
-        }
+    /// Instance of ConfigData
+    ConfigData _config;
 
-        /**
-            Starts the Options Menu
-        */
-        void options_menu()
-        {
-            Option options;
-            Audio *audio = new Audio();
-            bool has_background_music = false;
-            
-            while (!key_down(ESCAPE_KEY))
-            {
-                process_events();
-                clear_screen();
+    /// Vector of ConfigData
+    std::vector<ConfigData> _configs;
 
-                options.updateOption();
+    /// Vector of MenuButtons
+    std::vector<Button*> _menu_btns;
 
-                if(!has_background_music)
-                {
-                    audio->playMusic(options.getCurrentMusic(), options.getVolume());
-                    has_background_music=false;   
-                }
-                
-                if(options.isChangeMusic())
-                {
-                    has_background_music=false;
-                }
+    /// Vector of GameScreenButtons
+    std::vector<Button*> _game_btns;
+
+    /// Thoth Tech Company intro
+    Splashscreen _intro_thoth_tech;
+
+    /// Arcade Machine Team intro
+    Splashscreen _intro_arcade_team;
+
+    /// Arcade Machine Team developer names
+    std::vector<std::string> m_arcadeTeamDeveloperNames;
+
+    /// SplashKit Production intro
+    Splashscreen _intro_splashkit;
+
+    /// Main Menu Selector
+    Selector _selector_main_menu;
+
+    /// Games Menu Selector
+    Selector _selector_games_menu;
+
+    /// Instance of Grid
+    Grid _grid;
+
+    /// Mouse pointer
+    point_2d _mouse;
 
-                if(options.isChangeVoLume())
+    /// Button Action 
+    std::string _action;
+
+    /// Play clicked
+    bool _play_clicked = false;
+
+    /// Turn menu music on/off
+    bool _play_music = true;
+
+    /// Menu button node
+    ButtonNode *menu_button_node = nullptr;
+
+public:
+    // Default Constructor
+    ArcadeMachine()
+    {
+        // Instantiate fundamental objects
+        Helper helper;
+        ConfigData config;
+        Selector cursor("cursor");
+        Splashscreen intro_thoth_tech("intro_thoth_tech");
+
+        // load developer names into m_arcadeTeamDeveloperNames
+        // this can be a helper function later
+        {
```

Comment:
Not a huge rush for this feature, may as well create the function and move it where you think it should belong instead of rushing it into master unnecessarily.